samples/extensions/descriptor_buffer_basic/descriptor_buffer_basic.cpp

```diff
@@ -0,0 +1,471 @@
+	pipeline_create_info.pDynamicState                = &dynamic_state;
+
+	// We need to set this flag to let the implemenation know that this pipeline uses descriptor buffers
+	pipeline_create_info.flags = VK_PIPELINE_CREATE_DESCRIPTOR_BUFFER_BIT_EXT;
+
+	const std::array<VkPipelineShaderStageCreateInfo, 2> shader_stages = {
+	    load_shader("descriptor_buffer_basic/cube.vert", VK_SHADER_STAGE_VERTEX_BIT),
+	    load_shader("descriptor_buffer_basic/cube.frag", VK_SHADER_STAGE_FRAGMENT_BIT)};
+
+	pipeline_create_info.stageCount = static_cast<uint32_t>(shader_stages.size());
+	pipeline_create_info.pStages    = shader_stages.data();
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipeline));
+}
+
+// This function creates the desctiptor bffers and puts the descriptors into those buffers, so they can be used during command buffer creation
+void DescriptorBufferBasic::prepare_descriptor_buffer()
+{
+	// For sizing the descriptor buffers, we need to know the size for the descriptor set layouts the pipeline is using
+	std::array<VkDeviceSize, 2> descriptorLayoutSizes{};
+	vkGetDescriptorSetLayoutSizeEXT(get_device().get_handle(), descriptor_set_layout_buffer, &descriptorLayoutSizes[0]);
+	vkGetDescriptorSetLayoutSizeEXT(get_device().get_handle(), descriptor_set_layout_image, &descriptorLayoutSizes[1]);
+
+	// This buffer will contain resource descriptors for all the uniform buffers (one per cube and one with global matrices)
+	resource_descriptor_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                                 (static_cast<uint32_t>(cubes.size()) + 1) * descriptorLayoutSizes[0],
+	                                                                 VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT | VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT,
+	                                                                 VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	// Samplers and combined images need to be stored in a separate buffer due to different flags (VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT) (one image per cube)
+	image_descriptor_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                              static_cast<uint32_t>(cubes.size()) * descriptorLayoutSizes[1],
+	                                                              VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT | VK_BUFFER_USAGE_SAMPLER_DESCRIPTOR_BUFFER_BIT_EXT | VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT,
+	                                                              VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	// Put the descriptors into the above buffers
+	// This is done with vkGetDescriptorEXT
+	// We use pointers to offset and align the data we put into the descriptor buffers
+	VkDescriptorGetInfoEXT desc_info = {VK_STRUCTURE_TYPE_DESCRIPTOR_GET_INFO_EXT};
```

Comment:
Maybe it's easier to understand with separate `VkDescriptorGetInfoEXT`s for the buffers and the samplers?

Reply:
Yeah, that would make the code easier to follow. Will change that.